app/assets/javascripts/mtl/document_modal.coffee

```diff
@@ -0,0 +1,111 @@
+IMAGE_EXTENSIONS = ['png', 'jpg', 'gif', 'bmp']
+MTL.templates.document_modal = """
+  <div class="document-modal">
+    <div class="document-modal-dimmer"></div>
+    <div class="document-modal-header">
+      <div class="document-modal-title left white-text">
+        <%= MTL.fileIcon(filename, { class: 'left' }) %>
+        <%- filename %>
+      </div>
+      <div class="document-modal-toolbar right">
+        <a href="<%- url %>" target="_blank">
+          <%= MTL.icon('file_download', { class: 'white-text left' }) %>
+        </a>
+        <a href="#" data-mtl-document-modal="close">
+          <%= MTL.icon('close', { class: 'white-text left' }) %>
+        </a>
+      </div>
+
+      <div class="document-modal-content">
+      </div>
+
+      <% if(hasNext) { %>
+        <div class="document-modal-next btn black">
+          <%= MTL.icon('keyboard_arrow_right', { class: 'white-text' }) %>
+        </div>
+      <% } %>
+
+      <% if(hasPrev) { %>
+        <div class="document-modal-prev btn black">
+          <%= MTL.icon('keyboard_arrow_left', { class: 'white-text' }) %>
+        </div>
+      <% } %>
+    </div>
+  </div>
+"""
+
+MTL.templates.preview_unknown = """
+  <div class="no-preview">
+    <p class="center">
+      <%= MTL.icon('visibility_off', { size: 'large grey-text' }) %>
+    </p>
+    <a href="<%- url %>" target="_blank" class="btn btn-primary truncate">
+      <%= MTL.icon('file_download', { class: 'left' }) %>
+      <%- filename %>
+    </a>
+  </div>
+"""
+
+nextFileCard = ($fileCard) ->
+  $fileCard.next('[data-mtl-document-modal="open"]')
+
+prevFileCard = ($fileCard) ->
+  $fileCard.prev('[data-mtl-document-modal="open"]')
```

Comment:
Just one remark, right now, the next/prev arrows can be used for any documents on the page. It should actually be limited to the current file group they belong to.